### ISSUE TITLE
Retire deprecated features?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#13756]
Extending the request in issue #13749, the following are tests in test/deprecated that I _believe_ predated the 1.19 release, suggesting that we ought to be able to retire some code related to them for 1.20 (?).  I'm tagging these based on who added the test, which is not necessarily to say that person should / must retire the related feature (though any help in making sure it gets retired one way or another would be appreciated).

TODO:
- [ ] cast-to-c-string.chpl (@mppf)
- [ ] domain-range-member.chpl (@mppf)
- [ ] extern-classes.chpl (@benharsh)
- [ ] old-eigvals.chpl (@ben-albrecht)
- [ ] old-list.chpl (@benharsh)
- [ ] override-warning.chpl (@mppf)
- [ ] parentNameErr.chpl (@mppf on behalf of Preson)
- [ ] string-isEmptyString.chpl (@ben-albrecht)
- [ ] string-uchars-ulength-cp.chpl (@e-kayrakli on behalf of @dmk42)
- [ ] throw-borrowed.chpl (@mppf)

**Comment by mppf** [comment]
#13788 will remove test/deprecated/extern-classes

**Comment by mppf** [comment]
AFAIK domain-range-member.chpl should be removed in 1.21, or at least that was the plan when we deprecated domain.member

**Comment by bradcray** [comment]
> AFAIK domain-range-member.chpl should be removed in 1.21

I listed it based on its add date of 2018-10-17 which suggested to me it'd been deprecated by 1.19.  We typically remove things after one release of deprecation warnings (though we don't need to be strict about it, particularly in cases like this where we're not carrying a ton of code around to support it).

**Comment by mppf** [comment]
Right, but in discussion for that particular change, we planned to keep it deprecated for 2 releases instead of the usual 1. (I don't feel particularly strongly about this one way or the other, myself).

**Comment by dlongnecke-cray** [comment]
#13841 will remove the old `List` module.

**Comment by bradcray** [comment]
> we planned to keep it deprecated for 2 releases instead of the usual 1.

I'm not opposed to that, but don't remember the reason.  Was it a good one?  If we agree to do that, I think we can close this issue.  Otherwise, we're close!

**Comment by mppf** [comment]
I think it was just something along the lines of "since it's a trivial and annoying change", the best record I have of the discussion is https://github.com/chapel-lang/chapel/issues/5041#issuecomment-267385347 . I think we could remove it now if we wanted to (and additionally I think it's unlikely old code will continue to work for other reasons...)

**Comment by bradcray** [comment]
Closing this since 1.20 is wrapping up and we decided not to remove the last item whether through agreement or laziness.
